### ISSUE TITLE
[Paid Services - Documentation by Tarun Telang] MP API Tutorials

### ISSUE THREAD

**Issue by aeiras** [eclipse/microprofile-marketing#847]
Tracing history via MP forums
- [MicroProfile Tutorial Bid Proposal](https://groups.google.com/g/microprofile-marketing/c/CVfYp6-ozDw)
- [Funding MicroProfile Tutorial with 2023 Budget](https://groups.google.com/g/microprofile-marketing/c/MkZnAA2da5s)
- [
[microprofile-wg] MicroProfile tutorial update](https://www.eclipse.org/lists/microprofile-wg/msg02301.html)

**Comment by aeiras** [comment]
[
[Creating the Official MicroProfile Tutorial - Tarun Telang.pdf](https://github.com/eclipse/microprofile-marketing/files/13638309/Creating.the.Official.MicroProfile.Tutorial.-.Tarun.Telang.pdf)
Proposal for Creating MicroProfile Tutorial _ Werner Keil.pdf](https://github.com/eclipse/microprofile-marketing/files/13638308/Proposal.for.Creating.MicroProfile.Tutorial._.Werner.Keil.pdf)

**Comment by ttelang** [comment]
- Attended MP Marketing Call, 11th December 2023 - 11:30 pm -12:00 pm IST, as per request from Reza and Clark 
- Attached is the Presentation summarizing my Outline Proposal - 
[Outline for the Official MicroProfile Tutorial.pdf](https://github.com/eclipse/microprofile-marketing/files/13640112/Outline.for.the.Official.MicroProfile.Tutorial.pdf)
- I would be joining the upcoming MP Community call to get the above Outline further reviewed.
- As recommended in the MP Marketing Call, I would be following format (ASCIIdoc + Antora) as per Jakarta EE Tutorial - https://jakartaee.github.io/jakartaee-documentation/jakartaee-tutorial/current/index.html
- Awaiting contract from Eclipse Foundation to start the work, might need to reconsider the timeline as MPWG is going on holidays. I will update this ticket after signing the contract and will also share queries for the MP Marketing.

**Comment by aeiras** [comment]
Dear @ttelang, 

Thank you for your generosity in attending today's meeting and introducing yourself while enabling many Q&A. 

This ticket has now been assigned to you, and you can use it as you see fit. 👍  This community collaborated with another wonderful MicroProfiler on a similar written project  [2019- Technical White Paper Pro Bono by Lars RÖWEKAMP! #3](https://github.com/eclipse/microprofile-marketing/issues/3) that has about 86 comments. **Therefore, ASK AWAY!** 

**Below is the core summary, in addition to your notes above, of today's feedback provided by MicroProfilers:**

-  MP goes into End of the Year Vacation from December 16th to January 7th. 
<img width="691" alt="image" src="https://github.com/eclipse/microprofile-marketing/assets/10731224/3605d559-bf14-4095-a025-3d4d9a346042">


The milestones listed above should be postponed to ~ one month later to enable MP proactive feedback. 

- During the MP Holiday period, The Tutorial contract agreement needs to be signed directly with the EF team. Reza will connect you with the respective EF members via the MPWG forum. Those conversations with the EF are expected to be private. You are most welcome to use this ticket to let us know when the agreements are signed, and you are a GO to initiate your Research, if any, during the Holiday period. 

- [MPWG Charter](https://www.eclipse.org/org/workinggroups/microprofile-charter.php) should help you navigate your contributions and collaboration with the MP Community. As mentioned today, MP's Openness is 1 of the seven pillars that should be protected while adding value to the ecosystem. 

<img width="883" alt="image" src="https://github.com/eclipse/microprofile-marketing/assets/10731224/435088f0-3d58-435c-ba58-53b4bc9c76ae">

- Your Research on anything MP is most desirable during the MP Holiday. In MP, the source of truth is the Git Repo and the website. 

----

**Requests** @ttelang, 

1. Can you please provide your email? Why?

As a follow-up, I created a Google Drive space for you in the MP community folder under [MP>Content](https://drive.google.com/drive/u/0/folders/1kRrOzf0G7PtG8VJXKODgDpVNOUKSUxnR)

- [Tarun Teland - Folder ](https://drive.google.com/drive/u/0/folders/1y7bV3QN9N0nwMbmuXW7tc-auBrQ5IpGQ)
folder no contains your Tutorial Outline & Proposal
Yet, you could use it to house any other documentation as you see fit. The folder access is always Public View yet restricted Edit to protect the content. 

We need your email to give you Editor's access to own and manage the folder. 

---

**Comment by ttelang** [comment]
@aeiras - thanks for creating the Google Drive space. For granting edit permissions, my email id is: tarun.telang@gmail.com.

**Comment by ttelang** [comment]
- I am awaiting the contract, shared my mailing address to Dennis who is drafting the contract. 
-- 12/21 - reviewed draft contract, as a last step awaiting final version for sign off.  
**-- 01/06 - I have signed the contract**
- Could not join MP Community call as it was scheduled at 1 am (IST) my local time. 
- Reze has introduced me to Kito to provide me pointers regarding leveraging Jakarta EE Tutorial.
- Thanks @aeiras for sharing about the earlier project on MP Whitepaper (https://microprofile.io/resources/#white-paper) and related conversations. It definitely gave good clarity about the process for this project.  
- I also went through the MPWG Charter and [How to create OSS Content Steps](https://github.com/eclipse/microprofile-marketing/wiki/Pro-Bono:-OSS-Content-Creation-Steps-for-White-Paper,-Case-Study,-e-Book...) looking forward for collaborating on this project as per recommended processes wherever relevant for this project.

**Comment by ttelang** [comment]
I have shared following queries to Kito via email. While, I am awaiting response, others may also respond here in case any further inputs. 

------------

I see there are two formats used for Jakarte EE tutorials: 
(1) https://eclipse-ee4j.github.io/jakartaee-tutorial/ (Old one)
(2) https://jakartaee.github.io/jakartaee-documentation/jakartaee-tutorial/current/index.html (New one)

1. Is the second one (2) the right format for creating the new Microprofile tutorial ?

I am referring the following repositories and planning to create MicroProfile tutorial on similar lines: 
- https://github.com/jakartaee/jakartaee-tutorial
- https://github.com/eclipse-ee4j/jakartaee-examples
- https://github.com/jakartaee/jakartaee-documentation-ui/

From the above repos, I understand that I need to use the following tools to generate the MicroProfile tutorials:  
- AsciiDoc (https://asciidoc.org/) - Jan 8, 2024 - finished familiarizing with AsciiDoc Syntax
- Antora (https://antora.org/) - Started setting up Antora environment - I would be using Antora v3.7.1 for organizing the content for this site.

2. Is my understanding correct, kindly correct me or share any further details if I am missing out something.
3. Is there any recommended diagramming tool for creating architecture diagrams?
4. Is there any style guide for architecture diagrams that I should follow?

**Comment by aeiras** [comment]
> @aeiras - thanks for creating the Google Drive space. For granting edit permissions, my email id is [tarun.telang@gmail.com](mailto:tarun.telang@gmail.com).

Dear Tarun, 

Thank you for getting back to me so quickly. Your editor access has been sent this AM. Welcome to MPWG, contributing via Google Drive as well. :) 

---

I am glad the extra resources provided via this ticket are valuable to your initial MP research. 
You are already helping improve things in MP with the ticket you submitted. Amazing OSS form, Tarum. You got this! :) 

---

As we start the MP Holiday brake **up until** the 8th of January, I wish you a beautiful Holiday and a happy end of 2023!

**Comment by ttelang** [comment]
Following Items have been discussed so far: 

- I will be preparing the MicroProfile Tutorial as per the format -  
https://jakartaee.github.io/jakartaee-documentation/jakartaee-tutorial/current/index.html 

- The content would be written as per [Jakarta EE Tutorial Style Guide](https://github.com/jakartaee/jakartaee-tutorial/blob/main/STYLE_GUIDE.adoc)

- Diagrams will be created using draw.io as per the guidelines at this page - [Jakarta EE Tutorial - Creating Diagrams]- https://github.com/jakartaee/jakartaee-tutorial/blob/main/CONTRIBUTING.md#creating-diagrams.

- I am planning to store the assets of the MicroProfile Tutorials in the following repositories: 

| S.N | Repo | Purpose |
| ---|--------|--------|
| 1. | https://github.com/eclipse/microprofile-tutorial | Content of the MicroProfile Tutorial |
| 2. | https://github.com/eclipse/microprofile-documentation | Builds documentation site using the Antora Playbook.|
| 3. | https://github.com/eclipse/microprofile-documentation-ui | HTML and CSS assets used for the look and feel of the documentation site.|
| 4. | https://github.com/eclipse/microprofile-examples | Example source code used inside the Tutorial content| 

- I need to finalize whether 1&4 should be merged or kept separately.

- I would be referring [How Antora Works](https://docs.antora.org/antora/latest/how-antora-works/) to get started with how Antora orchestrator pulls up the content from different repos and generates the UI with content, search, etc.  

Thanks Kito for confirming the format and tools via email and sharing the style guides about writing content and creating diagrams. 

I have started organizing the content in this repository, https://github.com/ttelang/microprofile-tutorials - will get this copied to the new repository https://github.com/microprofile/microprofile-tutorials in a weeks time after it gets into decent shape.

**Comment by ttelang** [comment]
The tutorial will be based will be based on the currently available latest version of [MicroProfile 6.1](https://microprofile.io/compatible/6-1/). Let me know in case, anyone has any other opinion or views on this.

**Comment by ttelang** [comment]
Today (Jan 23, 2024), I attended MP Open API Technical Call today, presented the outline draft document - https://drive.google.com/file/d/1sSy5FyeQM_QYBPWK761XrBAtUtUQx6ye/view?usp=sharing for feedback on the MicroProfile OpenAPI Specification chapter. 

Below were some inputs I received: 
- Link for the recent API changes for MicroProfile OpenAPI post MicroProfile 5.0 
https://download.eclipse.org/microprofile/microprofile-open-api-3.1/microprofile-openapi-spec-3.1.html#api_changes_31
- For various technical issues or feedbacks, I would be sharing the drafts / outlines in this group - https://groups.google.com/g/microprofile

**Comment by Emily-Jiang** [comment]
@ttelang Thank you for joining the MP Technical call yesterday and providing an informative update! As promised on the call, when you are ready to get feedback, please email to the following 2 addresses: microprofile@googlegroups.com, microprofile-wg@eclipse.org

**Comment by Emily-Jiang** [comment]
This EF [issue](https://gitlab.eclipse.org/eclipsefdn/emo-team/emo/-/issues/674) was raised to get the dedicated microprofile-tutorial repo created.

**Comment by ttelang** [comment]
Thanks @Emily-Jiang for sharing about MicroProfile google group and working group mailing lists. I have subscribed to these and started a conversation regarding seeking feedback about the MicroProfile API Tutorial. I will engage with these forums for further inputs regularly.

**Comment by ttelang** [comment]
Please find below my revised plan for completing the MicroProfile APIs Tutorial: 

Milestone | Deliverables | Date | ETA
 -- | -- | -- | --
  |   |  
Design | Final Content Outline and Structure | Done  
  |   |  
Milestone 1 | Draft Completion for Chapters 1 | 6th Feb 2024 | Done
  | Draft Completion for Chapters 2 | 8th Feb 2024 | Done 
  | Draft Completion for Chapters 3 | 10th Feb 2024 | Done 
  |   |  
Milestone 2 | Draft Completion for Chapters 4 | 12th Feb 2024 | Done
  | Draft Completion for Chapters 5 | 14th Feb 2024 | Done
  | Draft Completion for Chapters 6 | 16th Feb 2024 | Done 
  |   |  
Milestone 3 | Draft Completion for Chapters 7 | 18th Feb 2024 | Done
  | Draft Completion for Chapters 8 | 20th Feb 2024 | 7th March
  | Draft Completion for Chapters 9 | 22th Feb 2024 | 7th March
  | Draft Completion for Chapters 10 | 24th Feb 2024 | 8th March 
  | Draft Completion for Chapter 11 | 26th Feb 2024 | 8th March 
  |   |  
Milestone 4 | Finalized Tutorial | 29th Feb 2024 | 10th March

**Comment by ttelang** [comment]
I am working on this repo - https://github.com/ttelang/microprofile-tutorials. After the new repo within the eclipse organization is created, I would move the content there.

Update - This is the new repo for MicroProfile tutorial - https://github.com/eclipse/microprofile-tutorial, please watch this for further updates.

**Comment by Emily-Jiang** [comment]
> This EF [issue](https://gitlab.eclipse.org/eclipsefdn/emo-team/emo/-/issues/674) was raised to get the dedicated microprofile-tutorial repo created.

Created [another](https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/4313) issue to chase up as I might have created the previous issue under a wrong team.

**Comment by jclingan** [comment]
I am worried about using the MicroProfile starter as the basis for the tutorial. It creates a mess regarding the variability of JDK/MicroProfile Version/Spec support. Some implementations, like Quarkus, [focus on its own starter](https://code.quarkus.io/) and not the MicroProfile starter because it is more feature-specific to Quarkus. 

Maybe a more appropriate approach is for each runtime to provide a CLI, pom.xml, or instructions for project creation. During the development of the tutorial, I suspect each implementation would have to tweak the instructions as we learn more.

**Comment by ttelang** [comment]
> I am worried about using the MicroProfile starter as the basis for the tutorial. It creates a mess regarding the variability of JDK/MicroProfile Version/Spec support. Some implementations, like Quarkus, [focus on its own starter](https://code.quarkus.io/) and not the MicroProfile starter because it is more feature-specific to Quarkus.
> 
> Maybe a more appropriate approach is for each runtime to provide a CLI, pom.xml, or instructions for project creation. During the development of the tutorial, I suspect each implementation would have to tweak the instructions as we learn more.

I attended MP Starter call on 12th Feb 2024. MP6.1 supports is not yet there. 

Below is what I am planning to cover. 

Introducing readers about generating a new project structure using MicroProfile Starter. Will also add a note like something like below: 

IMPORTANT: At the time of writing this tutorial, the latest MicroProfile released version was 6.1. We will not be using MicroProfile Starter to generate the project structure as it does not currently support this version.

**Comment by Emily-Jiang** [comment]
@ttelang the github repo https://github.com/eclipse/microprofile-tutorial has been created. You can create PRs to move your materials there.

**Comment by ttelang** [comment]
Kindly review and share your feedback, inputs or suggestions as comments directly to the Google Docs. Below is the list of chapters that are Ready for Review:
- [Front Matter](https://docs.google.com/document/d/1ZKZ_HGTGNApNh5zPztC6sIo9i4uQQo3Z4awux2_cNpg/edit?usp=sharing) - _Sections:  Legal, About this Tutorial, Who is this Tutorial for, What will be covered, Project, Downloading the Code, Prerequisites, Learning Objectives, Conventions_

- [Chapter 1: Introduction to MicroProfile](https://docs.google.com/document/d/1vfiuCjrFdaPb-PnN8kNgSZtvvHg4oLvmST-LcVAgE70/edit?usp=sharing) - _Sections: What is MicroProfile, Need for MicroProfile, MicroProfile Specifications, Current MicroProfile Implementations, Architecture Philosophy, Benefits of MicroProfile, Relationship with Jakarta EE Specification_

- [Chapter 2: Getting Started with MicroProfile](https://docs.google.com/document/d/1e37CLbD_ajm69TWSDNm6PXr9tlhMfdDwjV93RV1tr3A/edit?usp=sharing) -
_Sections: Development Environment Setup, Build Tools, Integrated Development Environments, Setting up MicroProfile Runtime, MicroProfile Starter, Creating a Java Project, Creating a REST Resource_

- [Chapter 3: Jakarta EE Core Profile](https://docs.google.com/document/d/1jMZup42HcD07p-cBoNY81z_LHx90zlw_-VEiRTWoOeE/edit?usp=sharing) - _Sections: Understanding the Jakarta EE 10 Core Profile, Key Specifications in Core Profile, Managing Component Dependencies, Handling HTTP Methods and Resources_

- [Chapter 4: MicroProfile Open API](https://docs.google.com/document/d/18q-Uw6T91kwoxDzbaUmLIY2s-HpT2jq2J2Rr_F7WHGA/edit?usp=drive_link) - _Sections: Introduction to MicroProfile OpenAPI, API Specification using MicroProfile Open API, Generating API Documentation, Exploring the APIs using Swagger UI_

- [Chapter 5: MicroProfile Configuration](https://docs.google.com/document/d/1FD_FimiNvZlMIFalB_zLIcQ-t9Wxp2ug8J2yQCG32cg/edit?usp=sharing) - _Sections: Understanding MicroProfile Configuration, Different environments required for Microservices development, Working with Various Configuration Sources, Key capabilities of MicroProfile Configuration, Implementing Configuration Properties, Creating a Custom Configuration Source, Dynamic Updates and Handling Configuration Change Events, Managing Configuration for Different Environments, Securing Configuration and Best Practices_

- [Chapter 6: MicroProfile Health](https://docs.google.com/document/d/1fc4-CSDGAtlnrAuofsM9A0BsUiVfSa799cJ4qY3EPYI/edit?usp=sharing) - _Sections: Overview of MicroProfile Health, Types of Health Checks, Standard Health Checks, Implementing and Exposing Health Checks, Logging and Reporting Health Checks, Best Practices for Effective Health Checks_

- [Chapter 7: MicroProfile Metrics](https://docs.google.com/document/d/1VAmdX_pX7oY6iCVNYC_3Fa3PpL-xtbfrK4lR27xzRmI/edit?usp=sharing) - _Sections: Introduction to MicroProfile Metrics, Need for Metrics in Microservices, Types of Metrics, Adding MicroProfile Metrics Dependency, Metrics Annotations, Using Metadata with Metrics, Categories of Metrics, Metric Registry, Instrumenting Microservices with Metrics, Creating Custom Metrics_

- [Chapter 8: MicroProfile Fault Tolerance](https://docs.google.com/document/d/1-kEN8A2fYxY1NehBObpwf0dALITC5aibpA4Ds2Y8Vl0/edit?usp=sharing) _Sections: What is Fault Tolerance?, Key Strategies for Enhancing Fault Tolerance, Implementing Retry Policies and Configuration, Avoiding and Managing Cascading Failures, Configuring Circuit Breaker, Setting Timeouts, Implementing Fallback Logic, Isolating Resources for Fault Tolerance_

- [Chapter 9: MicroProfile OpenTelemetry](https://docs.google.com/document/d/1afKCgrIsgMrBAzczLXJANmK0jArnb2mwh6pRsGVhkLM/edit?usp=sharing) _Sections: Introduction to MicroProfile OpenTelemetry, Tracing Concepts (Spans, Traces and Context Propagation), Instrumenting OpenTelemetry, Setting up Tracing  Providers, Context Propagation and Correlation, Analyzing Traces, Security Considerations for Tracing_

- [Chapter 10: MicroProfile JWT Authentication](https://docs.google.com/document/d/15LM3tr596KRRGirWBKKZKlQ6qCHDkK8W2FAmMUGoHTo/edit?usp=sharing) _Sections: Introduction to JWT Authentication, Understanding JSON Web Tokens, Request Flow, Obtaining and Validating JWT Tokens, Defining User Claims and Scopes in JWT, Role-Based Access Control (RBAC), Setting Token Expiry Times for Security, Error Handling, Best Practices for JWT Authentication_
 
- [Chapter 11: MicroProfile Rest Client](https://docs.google.com/document/d/1K1BRSf2R_7uxOf8k_NkY6K89_Nr4542lT2kA8UQOLr8/edit?usp=sharing) _Sections: Introduction to MicroProfile Rest Client, Setting up Dependencies, Defining a Rest Client Interface, Parameter Configuration, Request and Response Handling, Working with JSON Data format, Error Handling Strategies_

Above is the complete list of Drafts for all the chapters of the Official MicroProfile 6.1 Tutorial.

**Comment by ttelang** [comment]
> @ttelang the github repo https://github.com/eclipse/microprofile-tutorial has been created. You can create PRs to move your materials there.

@Emily-Jiang - Thanks for getting this repo created. I will start creating PRs in next couple of days for migrating the tutorial content. 

We would need 3 more repos as for the MicroProfile API tutorial as mentioned in the table here - https://github.com/eclipse/microprofile-marketing/issues/847#issuecomment-1864488393

**Comment by Emily-Jiang** [comment]
> > @ttelang the github repo https://github.com/eclipse/microprofile-tutorial has been created. You can create PRs to move your materials there.
> 
> @Emily-Jiang - Thanks for getting this repo created. I will start creating PRs in next couple of days for migrating the tutorial content.
> 
> We would need 3 more repos as for the MicroProfile API tutorial as mentioned in the table here - [#847 (comment)](https://github.com/eclipse/microprofile-marketing/issues/847#issuecomment-1864488393)

@ttelang we discussed this at the last Technical call. We agreed to just have one repo [microprofile-tutorial](https://github.com/eclipse/microprofile-tutorial). You can add multiple modules in that repo.

**Comment by Emily-Jiang** [comment]
@ttelang I have the MP Tutorial [repo](https://github.com/eclipse/microprofile-tutorial) set up for you with Apache License and it is ready for you to create PRs.

**Comment by ttelang** [comment]
> @ttelang I have the MP Tutorial [repo](https://github.com/eclipse/microprofile-tutorial) set up for you with Apache License and it is ready for you to create PRs.

Hi @Emily-Jiang , I have created the following PRs for your review:
- [9 - Preface (Front Matter)](https://github.com/eclipse/microprofile-tutorial/pull/9) - Merged ✅
- [10 - Chapter 1: Introduction to MicroProfile](https://github.com/eclipse/microprofile-tutorial/pull/10) - Merged ✅
- [11 - Chapter 2: Getting Started with MicroProfile](https://github.com/eclipse/microprofile-tutorial/pull/11) - Merged ✅
- [12 - Chapter 3: Jakarta EE 10 Core Profile](https://github.com/eclipse/microprofile-tutorial/pull/12) - Merged ✅
- [13 - Chapter 4: MicroProfile Open API](https://github.com/eclipse/microprofile-tutorial/pull/13) - Merged ✅
- [14 - Chapter 5: MicroProfile Configuration](https://github.com/eclipse/microprofile-tutorial/pull/14) - Merged ✅
- [15 - Chapter 6: MicroProfile Health]
(https://github.com/eclipse/microprofile-tutorial/pull/15) - Merged ✅
- [16 - Chapter 7: MicroProfile Metrics](https://github.com/eclipse/microprofile-tutorial/pull/16) - Merging ✅
- [17 - Chapter 8: MicroProfile Fault Tolerance] (https://github.com/eclipse/microprofile-tutorial/pull/17) - Merging ✅
- [18 - Chapter 9: MicroProfile Telemetry](https://github.com/eclipse/microprofile-tutorial/pull/18) - Merging ✅
- [20 - Source Code] (https://github.com/microprofile/microprofile-tutorial/pull/20)
- [25 - Chapter 10: MicroProfile JWT](https://github.com/microprofile/microprofile-tutorial/pull/25) - 🔄 Under Review
- [26 - Chapter 11: MicroProfile Rest Client](https://github.com/microprofile/microprofile-tutorial/pull/26) - 🔄 Under Review

**Comment by ttelang** [comment]
> I am worried about using the MicroProfile starter as the basis for the tutorial. It creates a mess regarding the variability of JDK/MicroProfile Version/Spec support. Some implementations, like Quarkus, [focus on its own starter](https://code.quarkus.io/) and not the MicroProfile starter because it is more feature-specific to Quarkus.
> 
> Maybe a more appropriate approach is for each runtime to provide a CLI, pom.xml, or instructions for project creation. During the development of the tutorial, I suspect each implementation would have to tweak the instructions as we learn more.

The tutorial will take a neutral approach avoid emphasizing any single MicroProfile implementation.
- Examples within the tutorial will be centered around MicroProfile specifications
- We would be sharing only about the minimal pom.xml containing only the MicroProfile dependency for the specifications covered in the corresponding chapter.
- If required, Instructions for using popular implementations (Open Liberty, WildFly, Payara etc.) would be covered in a separate section.
The complete code listing will be provided in separate repository for e.g. eclipse/microprofile-examples

Collaborating with the MicroProfile community, we can seek links to more in-depth guides provided by the implementations themselves.

**Comment by ttelang** [comment]
Recorded these videos 🎥 while verifying the content of first three chapters of MicroProfile Tutorial (Milestone 1) 

- [Installing JDK 17](https://youtu.be/MJNlW5_LNv0) ▶️ (2 mins)
- [Developing a Microservice using Jakarta EE + MicroProfile in GitHub Codespaces](https://youtu.be/O6ERkg5DHZ8) ▶️  (20 mins)
- [Building RESTful Web Services using Jakarta EE 10 Core Profile, MicroProfile Open API, MicroProfile Config, MicroProfile Health ](https://www.youtube.com/live/FWQo5s9gBCo) ▶️ (158 mins)

**Comment by jclingan** [comment]
@ttelang Do you have an update? It's been a while and we haven't seen updates on the Google Docs that we review. Are you waiting on an action item from us?

**Comment by croundy** [comment]
@ttelang  please provide a status update to this issue. Thanks!

**Comment by jclingan** [comment]
@ttelang We'd like to get an update before the steering committee call on Tuesday.

Thanks!

**Comment by jclingan** [comment]
@ttelang Ping. Any updates?

**Comment by rstjames** [comment]
Update from Marketing call Jan 6 2025

Tarun - MPWG Technical Tutorial - 
[[Paid Services - Documentation by Tarun Telang] MP API Tutorials #847](https://github.com/eclipse/microprofile-marketing/issues/847) 
[Chapter 8: MicroProfile Fault Tolerance](https://docs.google.com/document/d/1-kEN8A2fYxY1NehBObpwf0dALITC5aibpA4Ds2Y8Vl0/edit?usp=sharing) - Ready for Review
7 chapters are completed in the github
2 chapters are 80% done
2 chapters need to be written that makes up for the 20% of work left to do
11 chapters total
MP Holiday vacation starts on the 16th of December - We are back on January 6th.

**Comment by ttelang** [comment]
Status Update - 29th January 2025

✅ Chapters 01 - 08: Completed and available in GitHub.
🔄 Chapters 09 & 10: Under review. Currently incorporating feedback from the MP Community.
✍️ Chapter 11 (Final Chapter): Draft in progress. Planning to share for review in the coming days.

**Comment by ttelang** [comment]
Status Update – March 10, 2025

✅ Chapters 01 - 08: Completed and merged in GitHub.
✅  Chapter 09 – Ready for merge.
🔄  Chapter 10 – Incorporating review feedback.
✍️ Chapter 11 (Final Chapter) – Draft in progress. Would be sharing the draft for review this week.

**Comment by ttelang** [comment]
Status Update – March 21, 2025

✅ Drafts for all 11 Chapters completed.
✅ Chapters 01 - 09: Completed and merged in GitHub.
🔄 Chapter 10 & 11: Under Review in GitHub

**Comment by ttelang** [comment]
@croundy – Apologies for the delayed response and for overlooking your request for updates. 

I have completed drafting for all chapters of the tutorial, with 9 / 11  chapters already approved and committed to the GitHub repository. The remaining 2 chapters are currently under review by the MicroProfile community. 

After incorporating the review feedback, I’ll be consolidating everything into a tutorial site to wrap up the project.

**Comment by ttelang** [comment]
📅 Status Update – April 1, 2025

✅ Chapters 01–09: Completed and merged into GitHub.
🔍 Chapters 10 & 11: Finalized and ready to merge. Pending final review feedback.
🌐 Antora Site Generation: Completed. The demo is ready (please see screenshots below), and changes are in a draft PR. Will merge post-review.

![Image](https://github.com/user-attachments/assets/10a0f442-f4ca-43e3-8075-d92bac0bcbd7)

![Image](https://github.com/user-attachments/assets/7b5a6d32-04e4-4e12-abd0-2b9e495fcae4)

![Image](https://github.com/user-attachments/assets/57367c3d-3e6b-4431-bd82-c23b1b1944b1)

![Image](https://github.com/user-attachments/assets/811919de-9095-4c4f-8628-b871c64e1de2)

**Comment by ttelang** [comment]
📅 Status Update – April 8, 2025

✅ Update: Tutorials Aligned with MicroProfile Branding

MicroProfile tutorials have been updated to follow the official [MicroProfile Branding Guidelines](https://microprofile.io/resources/branding/).

Visual consistency verified across Light Theme and Dark Theme:

Light Theme
![Image](https://github.com/user-attachments/assets/7464dfa5-cfac-4274-8bf6-57f203a30683)

Dark Theme
![Image](https://github.com/user-attachments/assets/be7a5cdb-ecf7-4c05-a80f-31cbde9dfb0e)

📌 Awaiting 1: Review & Merge for the following Chapters from MicroProfile community:

- Chapter 10: MicroProfile JWT
- Chapter 11:  MicroProfile Rest Client

📌 Awaiting 2: New Repository 📦 Antora site UI code:
A dedicated repo is being prepared to host the UI code for the Antora site, in line with MicroProfile branding standards.
-  Proposed name: eclipse/microprofile-tutorial-ui

**Comment by ttelang** [comment]
📅 Status Update – April 28, 2025
All chapters are merged to the repo - https://github.com/microprofile/microprofile-tutorial

Next steps: 
- Merging the example code of the tutorial
- Merging the UI code for Antora site generation.

**Comment by ttelang** [comment]
@aeiras - You can review the MicroProfile Branded Antora UI Bundle at this place: 🔗 [Preview the Site](https://ttelang.github.io/microprofile-tutorial-ui/)

This UI bundle code for the tutorial site will be merged into this official repository:
💻 [GitHub Repository – microprofile-tutorial-ui](https://github.com/microprofile/microprofile-tutorial-ui)

**Comment by ttelang** [comment]
✅ All tasks completed – the MicroProfile Tutorial site is ready for publishing!

The following pull requests can now be safely merged to generate the full Tutorial content:
- https://github.com/microprofile/microprofile-tutorial-ui/pull/2
- https://github.com/microprofile/microprofile-tutorial/pull/20
- https://github.com/microprofile/microprofile-tutorial/pull/28

This marks the completion of the tutorial site setup using the Antora site generator and custom MicroProfile-branded UI bundle.

🚀 Looking forward to the final merge and public release!

**Comment by ttelang** [comment]
Here is the preview URL for the MicroProfile API Tutorial - https://ttelang.github.io/microprofile-tutorial/microprofile-tutorial/6.1/index.html for everyone's review. 

The following pull requests can now be safely merged to enable CI/CD via GitHub Actions  for the MicroProfile Tutorial content:
- https://github.com/microprofile/microprofile-tutorial/pull/33

Publishing the MicroProfile Tutorial (Profile) in MicroProfile site (microprofile.io) to be taken care by @MicroProfile Community Members
- Create a Resources -> Tutorial link 
- Configure Subdomain for MicroProfile GitHub Pages

**@aeiras - As per your advised during the MP Steering Committee call I have created an issue to track the above activity for Publishing the MicroProfile Tutorial - https://github.com/microprofile/microprofile-marketing/issues/1081**

Invoice shared with @jclingan, @Emily-Jiang to confirm the task completion and forwarding it for further processing.

**Comment by ttelang** [comment]
The MicroProfile API Tutorials are available at this URL - https://microprofile.github.io/microprofile-tutorial/microprofile-tutorial/6.1/index.html

**Comment by ttelang** [comment]
@aeiras, @Emily-Jiang, @jclingan - we can close this ticket.
#1081 - Please track this ticket for linking the MicroProfile Tutorial to the MicroProfile.io site. This ticket is still unassigned; please assign this to the web developer responsible for updating the site with the link to the MicroProfile Tutorial.

**Comment by ttelang** [comment]
Reminder: @aeiras, @Emily-Jiang, @jclingan, we can close this ticket.

I do not have the permissions to close this ticket.
